src/literal.rs

```diff
@@ -166,62 +173,77 @@ fn c_unicode_escape(n: Vec<u8>) -> Option<CChar> {
-                | map_opt!(many_m_n!(1, 3, octal), |v| c_raw_escape(v, 8))
-                | map_opt!(
-                    preceded!(char!('x'), many1!(hexadecimal)),
-                    |v| c_raw_escape(v, 16)
-                )
-                | map_opt!(
-                    preceded!(char!('u'), many_m_n!(4, 4, hexadecimal)),
-                    c_unicode_escape
-                )
-                | map_opt!(
-                    preceded!(char!('U'), many_m_n!(8, 8, hexadecimal)),
-                    c_unicode_escape
-                )
-        )
-    )
-);
-
-named!(
-    c_width_prefix,
-    alt!(tag!("u8") | tag!("u") | tag!("U") | tag!("L"))
-);
-
-named!(
-    c_char<CChar>,
-    delimited!(
-        terminated!(opt!(c_width_prefix), char!('\'')),
-        alt!(escaped_char | map!(byte!(0 ..= 91 /* \=92 */ | 93 ..= 255), CChar::from)),
-        char!('\'')
-    )
-);
-
-named!(
-    c_string<Vec<u8>>,
-    delimited!(
-        alt!(preceded!(c_width_prefix, char!('"')) | char!('"')),
-        fold_many0!(
-            alt!(
-                map!(escaped_char, |c: CChar| c.into())
-                    | map!(is_not!([b'\\', b'"']), |c: &[u8]| c.into())
+fn escaped_char(i: &[u8]) -> nom::IResult<&[u8], CChar> {
+    use nom::branch::alt;
+    use nom::character::complete::{char, one_of};
+    use nom::combinator::{map, map_opt};
+    use nom::multi::{many1, many_m_n};
+    use nom::sequence::preceded;
+    preceded(
+        char('\\'),
+        alt((
+            map(one_of(r#"'"?\"#), CChar::Char),
+            map(one_of("abfnrtv"), escape2char),
+            map_opt(many_m_n(1, 3, octal), |v| c_raw_escape(v, 8)),
+            map_opt(preceded(char('x'), many1(hexadecimal)), |v| {
+                c_raw_escape(v, 16)
+            }),
+            map_opt(
+                preceded(char('u'), many_m_n(4, 4, hexadecimal)),
+                c_unicode_escape,
+            ),
+            map_opt(
+                preceded(char('U'), many_m_n(8, 8, hexadecimal)),
+                c_unicode_escape,
             ),
+        )),
+    )(i)
+}
+
+fn c_width_prefix(i: &[u8]) -> nom::IResult<&[u8], &[u8]> {
+    use nom::branch::alt;
+    use nom::bytes::complete::tag;
+    alt((tag("u8"), tag("u"), tag("U"), tag("L")))(i)
+}
+
+fn c_char(i: &[u8]) -> nom::IResult<&[u8], CChar> {
+    use nom::branch::alt;
+    use nom::character::complete::char;
+    use nom::combinator::{map, opt};
+    use nom::sequence::{delimited, terminated};
+    delimited(
+        terminated(opt(c_width_prefix), char('\'')),
+        alt((
+            escaped_char,
+            map(byte!(0 ..= 91 /* \=92 */ | 93 ..= 255), CChar::from),
+        )),
+        char('\''),
+    )(i)
+}
+
+fn c_string(i: &[u8]) -> nom::IResult<&[u8], Vec<u8>> {
+    use nom::branch::alt;
+    use nom::bytes::complete::is_not;
+    use nom::character::complete::char;
+    use nom::combinator::map;
+    use nom::multi::fold_many0;
+    use nom::sequence::{delimited, preceded};
```

Comment:
Not a fan of all these uses at the start of functions. Is this canonical nom use now? I'd rather have a bunch of imports at the top.